src/main/java/transforms/SubstringTransform.java

```diff
@@ -4,30 +4,45 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.openlattice.shuttle.transformations.Transformation;
 import com.openlattice.shuttle.util.Constants;
+import java.util.Objects;
+
 
 public class SubstringTransform extends Transformation<String> {
     private final int index;
+    private final int endIndex;
 
     /**
      * Represents a transformation to get a substring starting from a certain index (eg 3th character: index = 2).
      *
      * @param index: where to start subsetting if prefix is found
+     * @param endIndex: where to end subsetting if prefix is found
      */
     @JsonCreator
     public SubstringTransform(
-            @JsonProperty( Constants.INDEX ) int index ) {
+            @JsonProperty( Constants.INDEX ) Integer index,
+            @JsonProperty( Constants.END_INDEX ) Integer endIndex ) {
         this.index = index;
+        this.endIndex = endIndex;
     }
 
-    @JsonProperty( Constants.LOC )
-    public int getSubstringLocation() {
+    @JsonProperty( Constants.INDEX )
+    public int getIndex() {
```

Comment:
This should return an `Integer` not `int` otherwise you'll NPE if `endIndex` is null

Also, have you tested this transformation on not passing `endIndex`? I feel like we've seen issues in the past where you end up forced to do 
```
{
  index: <blah>,
  endIndex: null
}
```
instead of just 
```
{
   index: <blah>
}
```
unless the type is set to `Optional<Integer>`, but I might be remembering wrong